klippa/src/os2.rs

```diff
@@ -0,0 +1,125 @@
+//! impl subset() for OS/2
+use crate::SubsetFlags;
+use crate::{Plan, SubsetError, SubsetError::SubsetTableError};
+use std::cmp::Ordering;
+use write_fonts::{
+    read::{
+        collections::IntSet,
+        tables::os2::{Os2, OS2_UNICODE_RANGES},
+        FontRef, TableProvider, TopLevelTable,
+    },
+    FontBuilder,
+};
+
+pub fn subset_os2(
+    font: &FontRef,
+    plan: &Plan,
+    builder: &mut FontBuilder,
+) -> Result<(), SubsetError> {
+    let os2 = font.os2().or(Err(SubsetTableError(Os2::TAG)))?;
+    let mut out = os2.offset_data().as_bytes().to_owned();
+
+    let us_first_char_index: u16 = plan.unicodes.first().unwrap_or(0xFFFF).min(0xFFFF) as u16;
+    out.get_mut(64..66)
+        .unwrap()
+        .copy_from_slice(&us_first_char_index.to_be_bytes());
+
+    let us_last_char_index: u16 = plan.unicodes.last().unwrap_or(0xFFFF).min(0xFFFF) as u16;
+    out.get_mut(66..68)
+        .unwrap()
+        .copy_from_slice(&us_last_char_index.to_be_bytes());
+
+    if plan
+        .subset_flags
+        .contains(SubsetFlags::SUBSET_FLAGS_NO_PRUNE_UNICODE_RANGES)
+    {
+        builder.add_raw(Os2::TAG, out);
+        return Ok(());
+    }
```

Comment:
If anyone adds code below this they expect to execute the return may surprise them, plus you duplicate adding to the builder. Could you do it the other way around, as in if plan doesn't contain no prune update ranges, no early return?